.github/workflows/publish-extension.yml

```diff
@@ -0,0 +1,23 @@
+name: Publish STAC Extension JSON Schema
+on:
+  release:
+    types: [published]
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/extension')
+    steps:
+      - name: Inject env variables
+        uses: rlespinasse/github-slug-action@v4.4.1
+      - uses: actions/checkout@v4
+      - name: Parse extension version
+        run: |
+          extension_version=$(echo "$env.GITHUB_REF_SLUG" | sed 's/extension-//')
+          echo "EXTENSION_VERSION=${extension_version}" >> $GITHUB_ENV
+      - name: deploy JSON Schema for version ${{ env.EXTENSION_VERSION }}
+        uses: peaceiris/actions-gh-pages@v3.9.3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: json-schema
+          destination_dir: ${{ env.EXTENSION_VERSION }}
```

Comment:
Github now has a official pages action, which might be preferable over this custom one: https://github.com/actions/deploy-pages

Reply:
👀 I'll give it a shot, definitely preferable to a random action. It'll need a bit more steps but I think I can get it to do the right thing

Reply:
Added in https://github.com/hotosm/stactools-hotosm/pull/6/commits/ed13b2616428f75f118b61f5ae369e9345c39464

I'm 73% sure I'll need another iteration after seeing what CI does, but I think I got the right steps,

1. Figure out extension version from tag
2. Copy JSONSchema into a version specific subfolder within some "publish directory" 
3. Upload the entire "publish directory" as an artifact
4. Publish the artifact